plugins/typescript.js

```diff
@@ -236,6 +238,8 @@ module.exports = {
     '@typescript-eslint/prefer-ts-expect-error': 2,
     // enforce the consistent use of either backticks, double, or single quotes
     '@typescript-eslint/quotes': stylisticIssues.quotes,
+    // disallow members of unions and intersections that do nothing or override type information
+    '@typescript-eslint/no-redundant-type-constituents': 1,
```

Comment:
I would encourage this to be a `2` error, as it avoids code slop. Also, because this rule requires type information, it must move from here into the `typescript-semantics.js` file. This allows a consumer to selectively add these rules when the `@typescript-eslint/parser` is available (which can be too expensive for larger project linting performance).